packages/flutter/test/material/popup_menu_theme_test.dart

```diff
@@ -0,0 +1,255 @@
+
+    final AnimatedDefaultTextStyle text =
+        tester.widget<AnimatedDefaultTextStyle>(
+      find
+          .descendant(
+            of: find.byKey(popupItemKey),
+            matching: find.byType(AnimatedDefaultTextStyle),
+          )
+          .last,
+    );
+    expect(text.style, popupMenuEntryTheme.textStyle);
+  });
+
+  testWidgets('PopupMenuEntry widget properties take priority over theme',
+      (WidgetTester tester) async {
+    final PopupMenuEntryThemeData popupMenuEntryTheme = _popupMenuEntryTheme();
+    final Key popupButtonKey = UniqueKey();
+    final Key popupButtonApp = UniqueKey();
+    final Key popupItemKey = UniqueKey();
+
+    const Color surfaceContainerColor = Colors.purple;
+    const ShapeBorder shape = RoundedRectangleBorder(
+      borderRadius: BorderRadius.all(Radius.circular(9.0)),
+    );
+    const double elevation = 7.0;
+    const TextStyle textStyle = TextStyle(
+        color: Color(0x00000000), textBaseline: TextBaseline.alphabetic);
+
+    await tester.pumpWidget(MaterialApp(
+      theme: ThemeData(popupMenuEntryTheme: popupMenuEntryTheme),
+      key: popupButtonApp,
+      home: Material(
+        child: Column(
+          children: <Widget>[
+            PopupMenuButton<void>(
+              key: popupButtonKey,
+              elevation: elevation,
+              color: surfaceContainerColor,
+              shape: shape,
+              itemBuilder: (BuildContext context) {
+                final List<PopupMenuEntry<Object>> list = <PopupMenuEntry<Object>>[
+                  PopupMenuItem<void>(
+                    key: popupItemKey,
+                    textStyle: textStyle,
+                    child: const Text(''),
+                  ),
+                ];
+                return list;
+              },
+            ),
+          ],
+        ),
+      ),
+    ));
+
+    await tester.tap(find.byKey(popupButtonKey));
+    await tester.pumpAndSettle();
+
+    final Material button = tester.widget<Material>(
+      find
+          .descendant(
+            of: find.byKey(popupButtonApp),
+            matching: find.byType(Material),
+          )
+          .last,
+    );
+    expect(button.color, surfaceContainerColor);
+    expect(button.shape, shape);
+    expect(button.elevation, elevation);
+
+    final AnimatedDefaultTextStyle text =
+        tester.widget<AnimatedDefaultTextStyle>(
+      find
+          .descendant(
+            of: find.byKey(popupItemKey),
+            matching: find.byType(AnimatedDefaultTextStyle),
+          )
+          .last,
+    );
+    expect(text.style, textStyle);
+  });
+}
+
+PopupMenuEntryThemeData _popupMenuEntryTheme() {
+  return PopupMenuEntryThemeData(
+    surfaceContainerColor: Colors.orange,
+    shape: BeveledRectangleBorder(borderRadius: BorderRadius.circular(12)),
+    elevation: 12.0,
+    textStyle: const TextStyle(
+        color: Color(0xffffffff), textBaseline: TextBaseline.alphabetic),
```

Comment:
Specifying the baseline doesn't seem necessary and can be omitted

Reply:
Done

Reply:
textBaseline is null if left unspecified, and [basic.dart](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/basic.dart) requires that textBaseline be non-null.